File: src/main/CompletedPrograms/rockPaperScissors/Game.java
package rockPaperScissors;

import java.util.Random;
import java.util.Scanner;

public class Game {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] choices = {"", "Rock", "Scissors", "Paper"};

        while (true) {
            System.out.println("Choose: 1 - Rock, 2 - Scissors, 3 - Paper, 0 - Quit the game");
            int userChoice = scanner.nextInt();

            if (userChoice == 0) {
                break;
            }

            int computerChoice = new Random().nextInt(choices.length - 1) + 1;

            System.out.println("Your choice: " + choices[userChoice]);
            System.out.println("Computer's choice: " + choices[computerChoice]);

            int result = determineWinner(userChoice, computerChoice);

            switch (result) {
                case 0:
                    System.out.println("It's a tie!");
                    break;
                case 1:
                    System.out.println("You win!");
                    break;
                case -1:
                    System.out.println("You lose!");
                    break;
            }
        }

        System.out.println("Game over.");
    }

    private static int determineWinner(int userChoice, int computerChoice) {
        if (userChoice == computerChoice) {
            return 0; // It's a tie
        }

        return switch (userChoice) {
            case 1 -> (computerChoice == 2) ? 1 : -1; // Rock
            case 2 -> (computerChoice == 3) ? 1 : -1; // Scissors
            case 3 -> (computerChoice == 1) ? 1 : -1; // Paper
            default -> throw new IllegalArgumentException("Invalid choice");
        };
    }
}

File: src/main/CompletedPrograms/rockPaperScissors/Game.kt
// Rock, Paper, Scissors Game Code
package rockPaperScissors

import java.util.*

object Game {
    @JvmStatic
    fun main(args: Array<String>) {
        val scanner = Scanner(System.`in`)
        val choices = arrayOf("", "Rock", "Scissors", "Paper")
        while (true) {
            println("Choose: 1 - Rock, 2 - Scissors, 3 - Paper, 0 - Quit the game")
            val userChoice = scanner.nextInt()
            if (userChoice == 0) {
                break
            }
            val computerChoice = Random().nextInt(choices.size - 1) + 1
            println("Your choice: " + choices[userChoice])
            println("Computer's choice: " + choices[computerChoice])
            val result = determineWinner(userChoice, computerChoice)
            when (result) {
                0 -> println("It's a tie!")
                1 -> println("You win!")
                -1 -> println("You lose!")
            }
        }
        println("Game over.")
    }

    private fun determineWinner(userChoice: Int, computerChoice: Int): Int {
        return if (userChoice == computerChoice) {
            0 // It's a tie
        } else when (userChoice) {
            1 -> if (computerChoice == 2) 1 else -1
            2 -> if (computerChoice == 3) 1 else -1
            3 -> if (computerChoice == 1) 1 else -1
            else -> throw IllegalArgumentException("Invalid choice")
        }
    }
}
